### PR TITLE
Add method to check if a country uses postal codes

### PR DESCRIPTION
**Pull request by lucasuyezu**
This logic already lives internally in Shopify. Moving it here so it can be reused.

Review @wvanbergen @disaacs
cc @Shopify/shipping

### CODE REVIEW

File: test/unit/country_test.rb [hunk 1]
```diff
@@ -65,4 +65,13 @@ def test_country_names_are_alphabetized
...
   end
 
+  def test_ca_uses_postal_code
+    canada = Country.find('CA')
+    assert canada.uses_postal_code?
+  end
+
+  def test_qa_does_not_use_postal_code
+    canada = Country.find('QA')
+    refute canada.uses_postal_code?
```
Comment (disaacs) [inline test/unit/country_test.rb]: Use `qatar` as the variable name, just to be clear.

File: lib/active_utils/country.rb [hunk 2]
```diff
@@ -313,6 +313,20 @@ def to_s
       { :alpha2 => 'AX', :name => 'Åland Islands', :alpha3 => 'ALA', :numeric => '248' }
     ]
 
+    # Does this country use postal codes?  Default is true.
+    COUNTRY_USES_POSTALCODE = %w(
+      QA BZ BS BF BJ AG AE AI AO AW HK
+      FJ ML JM ZW YE UG TV TT TG TD PA
+      CW
+    ).inject(Hash.new(true)) do |map, country|
```
Comment (girasquid) [inline lib/active_utils/country.rb]: This seems like a lot more work than necessary - would this work?
```ruby
`!COUNTRIES_THAT_DONT_USE_POSTAL_CODES.include?(code(:alpha2).value)`
```

File: test/unit/country_test.rb (hunk 1 above)
Comment (lucasuyezu) [inline test/unit/country_test.rb]: Just changed `CA` to `Canada` and `QA` to `Qatar`.

File: lib/active_utils/country.rb (hunk 2 above)
Comment (lucasuyezu) [inline lib/active_utils/country.rb]: It would. One could argue that hash lookup is faster than `Array#include?` but this is such a small array. I've changed to your form.

File: lib/active_utils/country.rb [hunk 3]
```diff
@@ -313,6 +313,17 @@ def to_s
       { :alpha2 => 'AX', :name => 'Åland Islands', :alpha3 => 'ALA', :numeric => '248' }
     ]
 
+    # Does this country use postal codes?  Default is true.
```
Comment (girasquid) [inline lib/active_utils/country.rb]: Do you need this comment? Is it still valid?

Comment (lucasuyezu) [inline lib/active_utils/country.rb]: Nope. Removed.

**Comment by lucasuyezu** [comment]
@girasquid Is it ok to merge now?

**Comment by girasquid** [comment]
changelog message is a bit wordy (do you *really* need to explain why you added a method called `uses_postal_code?`) and I'd get another review to be sure, but code looks fine.

**Comment by garethson** [comment]
Changes LGTM. Changlog can probably just be `Adds uses_postal_code?`.

**Comment by lucasuyezu** [comment]
@girasquid @garethson Done, changed to just `Add #uses_postal_code? to ActiveUtils::Country`.

**Comment by RichardBlair** [comment]
:+1: shipit.

**Comment by girasquid** [comment]
:+1:

**Comment by disaacs** [comment]
:+1:
